File: app/src/main/java/com/owncloud/android/utils/glide/StringModelLoaderFactory.java
/*
 * Nextcloud - Android Client
 *
 * SPDX-FileCopyrightText: 2025 Your Name <your@email.com>
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

package com.owncloud.android.utils.glide;

import com.bumptech.glide.load.model.ModelLoader;
import com.bumptech.glide.load.model.ModelLoaderFactory;
import com.bumptech.glide.load.model.MultiModelLoaderFactory;

import java.io.InputStream;

import androidx.annotation.NonNull;

public class StringModelLoaderFactory implements ModelLoaderFactory<String, InputStream> {

    @NonNull
    @Override
    public ModelLoader<String, InputStream> build(@NonNull MultiModelLoaderFactory unused) {
        return new GlideStringStreamLoader();
    }

    @Override
    public void teardown() {
        // Do nothing.
    }
}

File: app/src/main/java/com/owncloud/android/utils/glide/StringModelLoaderFactory.kt
/*
 * Nextcloud - Android Client
 *
 * SPDX-FileCopyrightText: 2025 Alper Ozturk <alper.ozturk@nextcloud.com>
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */
package com.owncloud.android.utils.glide

import com.bumptech.glide.load.model.ModelLoader
import com.bumptech.glide.load.model.ModelLoaderFactory
import com.bumptech.glide.load.model.MultiModelLoaderFactory
import java.io.InputStream

class StringModelLoaderFactory : ModelLoaderFactory<String, InputStream> {
    override fun build(unused: MultiModelLoaderFactory): ModelLoader<String, InputStream> {
        return GlideStringStreamLoader()
    }

    override fun teardown() {
        // Do nothing.
    }
}
